go-controller/pkg/retry/obj_retry.go

```diff
@@ -755,3 +761,41 @@ func (r *RetryFramework) WatchResourceFiltered(namespaceForFilteredHandler strin
 
 	return handler, nil
 }
+
+// getPendingPodsForNetwork returns pods from netInfo namespaces that are in the Pending state
+func getPendingPodsForNetwork(kubeClient kube.InterfaceOVN, netInfo util.NetInfo) ([]*kapi.Pod, error) {
+	var allPods []*kapi.Pod
+
+	for _, namespace := range netInfo.GetNamespaces() {
+		pods, err := kubeClient.GetPods(namespace, metav1.ListOptions{
+			FieldSelector: fields.OneTermEqualSelector("status.phase", string(kapi.PodPending)).String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+		allPods = append(allPods, pods...)
+	}
+	return allPods, nil
+}
+
+// RequeuePendingPodsForNetwork enqueues all Pending pods from the namespaces associated with the given netInfo into the retryPods.
+func RequeuePendingPodsForNetwork(kubeClient kube.InterfaceOVN, netInfo util.NetInfo, retryPods *RetryFramework) error {
```

Comment:
how will this work for pods located on other namespaces, but attempting to attach themselves to a network defined on a shared namespace ? For instance, the `default` namespace ?

Reply:
I wasn't aware that this was possible. Modified the code to handle all pending pods.